gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -155,6 +155,9 @@
 	@Setter
 	private RubricsService rubricsService;
 
+	@Setter @Getter
+	private boolean clearUiSettingsAfterImport = false;
```

Comment:
This variable is visible to all threads and it appears as though it is holding some state specific to a particular request? If so then this is not a good location for setting something like this.

Reply:
Where is a good location?

Reply:
@ern SiteImport is a background thread (e.g., "pool-6-thread-1") while Wicket is storing on a user thread (e.g., "http-nio-8080-exec-2") .... how do you communicate from the background thread to the user thread that the wicket session storage should be cleared?

Reply:
IMO the move is to remove this data from the session.